api/common/request_id_util.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import (
+	"context"
+
+	"github.com/fnproject/fn/api/id"
+)
+
+const (
+	maxLength = 32
+)
+
+// FnRequestID returns the passed value if that is not empty or too long otherwise it generates a new unique ID
+func FnRequestID(ridFound string) string {
+	if ridFound == "" {
+		return id.New().String()
+	}
+	if len(ridFound) > maxLength {
+		// we truncate the rid to the value specified by maxLenght const
+		return ridFound[:maxLength]
```

Comment:
you don't like this and @rdallman  doesn't either so I am thinking that I am wrong :) As the request ID can be passed in the incoming request, I wanted to have a way to guard against insane big values passed. Looking at the HTTP RFC there is no limit on the length of a header field but the common pattern here is that the check for the size of headers is a webserver's job and in fact, most of them have default limits...Well yes you have convinced me to get rid of the truncation.

Reply:
Yes, a production server must go through configuring proper gin/http-server settings which includes header size, how long we wait for this header, I/O read/write timeouts, etc. etc. We can assume this will be covered unless we want to clamp on a specific size (eg. 2k) and return an error. Either way is fine...